coral/src/app/features/topics/schema-request/TopicSchemaRequest.test.tsx

```diff
@@ -0,0 +1,587 @@
+  { name: "DEV", id: "1" },
+  { name: "TST", id: "2" },
+  { name: "INFRA", id: "3" },
+];
+const mockedGetSchemaRegistryEnvironments = transformEnvironmentApiResponse(
+  mockedEnvironments.map((entry) => {
+    return createMockEnvironmentDTO(entry);
+  })
+);
+
+const fileName = "my-awesome-schema.avsc";
+const testFile: File = new File(["{}"], fileName, {
+  type: "image/jpeg",
+});
+
+describe("TopicSchemaRequest", () => {
+  const getForm = () => {
+    return screen.getByRole("form", {
+      name: `Request a new schema`,
+    });
+  };
+
+  afterEach(() => {
+    jest.resetAllMocks();
+  });
+
+  describe("handles loading and update state", () => {
+    beforeAll(() => {
+      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+      //@ts-ignore
+      useQuerySpy.mockReturnValue({ data: [], isLoading: true });
+      mockGetSchemaRegistryEnvironments.mockResolvedValue([]);
+      mockCreateSchemaRequest.mockImplementation(jest.fn());
+      customRender(<TopicSchemaRequest topicName={testTopicName} />, {
+        queryClient: true,
+      });
+    });
+
+    afterAll(() => {
+      cleanup();
+      useQuerySpy.mockRestore();
+    });
+
+    it("shows a form to request a schema", () => {
+      const form = getForm();
+
+      expect(form).toBeVisible();
+    });
+
+    it("shows a loading element while environments are being fetched", () => {
+      const form = getForm();
+      const loadingEnvironments = within(form).getByTestId(
+        "environments-select-loading"
+      );
+
+      expect(loadingEnvironments).toBeVisible();
+    });
+  });
+
+  describe("renders all necessary elements", () => {
+    beforeAll(async () => {
+      mockGetSchemaRegistryEnvironments.mockResolvedValue(
+        mockedGetSchemaRegistryEnvironments
+      );
+      mockCreateSchemaRequest.mockImplementation(jest.fn());
+      customRender(<TopicSchemaRequest topicName={testTopicName} />, {
+        queryClient: true,
+      });
+      await waitForElementToBeRemoved(
+        screen.getByTestId("environments-select-loading")
+      );
+    });
+
+    afterAll(() => {
+      cleanup();
+    });
+
+    it("shows a form to request a schema", () => {
+      const form = getForm();
+
+      expect(form).toBeVisible();
+    });
+
+    it("shows a required select element to choose an environment", () => {
+      const form = getForm();
+      const select = within(form).getByRole("combobox", {
+        name: /Select environment/i,
+      });
+
+      expect(select).toBeEnabled();
+      expect(select).toBeRequired();
+    });
+
+    it("renders all options for environment based on api data", () => {
+      const form = getForm();
+      const select = within(form).getByRole("combobox", {
+        name: /Select environment/i,
+      });
+      const options = within(select).getAllByRole("option");
+
+      expect(options).toHaveLength(mockedEnvironments.length + 1);
+    });
+
+    mockedEnvironments.forEach((env) => {
```

Comment:
I wonder if using [test.each](https://jestjs.io/docs/api#testeachtablename-fn-timeout) would only yield syntactic benefits (is _syntactic_ even a word ?)

Reply:
After trying to use it a few times in the early days, but I've fallen into the habit of using `.forEach` because I feel like it makes tests more readable. But I feel I could try that again, with proper variable names it could work nicely. 🤔  

Let me see...

Reply:
```
    mockedEnvironments.forEach((env) => {
      it(`renders a option ${env.name} with the environments id set as value ${env.id}`, () => {
        const form = getForm();
        const select = within(form).getByRole("combobox", {
          name: /Select environment/i,
        });
        const option = within(select).getByRole("option", { name: env.name });

        expect(option).toBeVisible();
        expect(option).toHaveValue(env.id);
      });
    });
```
vs 

```
    test.each(mockedEnvironments)(
      `renders a option $name with the environments id set as value $id`,
      ({ name, id }) => {
        const form = getForm();
        const select = within(form).getByRole("combobox", {
          name: /Select environment/i,
        });
        const option = within(select).getByRole("option", { name: name });

        expect(option).toBeVisible();
        expect(option).toHaveValue(id);
      }
    );
```

I think that's ok from readability? It still looks a bit off to me, but that could be because it's unfamiliar. Maybe a good idea to revisit some of my resting habits :D